app/Models/User.php

```diff
@@ -17,6 +22,111 @@ class User extends Authenticatable
     use HasProfilePhoto;
     use Notifiable;
     use TwoFactorAuthenticatable;
+    use HasRoles;
+
+    /**
+     * Establishes a relationship between the user and the company they're part of
+     * (if they have a company)
+     * @return BelongsTo
+     */
+    public function company(): BelongsTo
+    {
+        return $this->belongsTo(Company::class);
+    }
+
+    /**
+     * Hides a many-to-many relationship with presentations
+     * and implements relationship with linking table UserPresentation
+     * Please don't use this, instead refer to the methods below
+     * @return HasMany
+     */
+    public function userPresentations(): HasMany
+    {
+        return $this->hasMany(UserPresentation::class);
+    }
+
+    /**
+     * Assign the user to a presentation, based on the role that
+     * was passed - participant or speaker
+     * Returns true if the user successfully was added to the presentation with their role;
+     * Returns false if the user wasn't attached to the presentation
+     * @param $presentation
+     * @param string $role
+     * @return void
+     */
+    public function joinPresentation($presentation, string $role = 'participant'): bool
+    {
+        if ($this->speaker) {
+            // The user is already a speaker of another presentation
+            if ($role == 'speaker') {
+                return false;
+            }
+
+            // The user is a speaker of this presentation, and cannot be a participant
+            if ($this->speaker->id == $presentation->id && $role == 'participant') {
+                return false;
+            }
+        }
+
+        // The user is already enrolled as a participant in this presentation
+        if ($this->participant->contains($presentation)) {
+            return false;
+        }
+
+        UserPresentation::create([
+            'user_id' => $this->id,
+            'presentation_id' => $presentation->id,
+            'role' => $role
+        ]);
+
+        return true;
+    }
+
+    /**
+     * Dis-enrol participant from a presentation
```

Comment:
```suggestion
     * Unenroll participant from a presentation
```
Nitpicky, but this is the correct grammar.